apps/labs/posts/implement-lapack-routines-in-stdlib.md

```diff
@@ -0,0 +1,210 @@
+        </p>
+    </td>
+  </tr>
+</table>
+
+## Challenges during Fortran to JS conversion
+
+1. Supporting both `row-major` and `column-major`.
+
+
+<table>
+  <tr>
+    <td style="vertical-align: top; width: auto;">
+        <p>
+            Fortran stores array elements in a column-major format, unlike C or JavaScript, which prefer row-major storage. Following the approach used in LAPACKE, we decided to introduce a new parameter, order, in each implementation to specify the storage layout. Based on the value of order, there would be distinct implementations and optimizations for each layout.
+        </p>
+        <p>
+            The order we loop through multidimensional arrays can have a big impact on speed. Fortran is as said column-major, Meaning consecutive elements of a column are stored next to each other in memory, and we should loop through arrays in this order order of columns unlike conventional looping over rows.
+        </p>
+    </td>
+    <td style="vertical-align: top; width: auto;">
+      <img src="/posts/implement-lapack-routines-in-stdlib/image-3.png" alt="alt text" style="width:3200px;height:300px;">
+    </td>
+  </tr>
+</table>
+
+Let's illustrate this with an example. Consider a 2D array `A` of size `5x4`. To iterate over the array in row-major order, we would loop through the rows first, followed by the columns. In contrast, to iterate over the array in column-major order, we would loop through the columns first, followed by the rows. The code snippet below demonstrates this concept.
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+var A; var B; var i; var j;
+
+/**
+* 5x4 matrix
+* [  1,  2,  3,  4 ],
+* [  5,  6,  7,  8 ],
+* [  9, 10, 11, 12 ],
+* [ 13, 14, 15, 16 ],
+* [ 17, 18, 19, 20 ]
+*/
+
+// row major
+A = new Float64Array( [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20 ] );
+
+// column major
+B = new Float64Array( [ 1, 5, 9, 13, 17, 2, 6, 10, 14, 18, 3, 7, 11, 15, 19, 4, 8, 12, 16, 20 ] );
+
+// iterate over A in cache optimal order
+for ( i = 0; i < 5; i++ ) {
+	for ( j = 0; j < 4; j++ ) {
+		console.log( A[ ( i * 4 ) + j ] );
+	}
+}
+
+// iterate over B in cache optimal order
+for ( i = 0; i < 4; i++ ) {
+	for ( j = 0; j < 5; j++ ) {
+		console.log( B[ ( i * 5 ) + j ] );
+	}
+}
+```
+
+Thereby, we need to ensure that our implementations are optimized for both row-major and column-major orders. We employ various optimization techniques, such as loop tiling and cache optimization, to enhance performance. While some of these optimizations are already present in Fortran codes, simplifying the translation process, in most cases, we need to identify and implement these optimizations ourselves to achieve optimal performance.
```

Comment:
It would be good to show before and after here. Go from an implementation without, e.g., loop interchange to an implementation with loop interchange.

Reply:
A benchmark plot would be beneficial.